src/base64.h

```diff
@@ -84,11 +85,15 @@ size_t base64_decode_fast(char* const dst, const size_t dstlen,
   size_t i = 0;
   size_t k = 0;
   while (i < max_i && k < max_k) {
-    const uint32_t v =
-        unbase64(src[i + 0]) << 24 |
-        unbase64(src[i + 1]) << 16 |
-        unbase64(src[i + 2]) << 8 |
-        unbase64(src[i + 3]);
+    // TODO(@juanarbol): use cares_get_32bit here.
+    const unsigned char txt[4] = {
+      static_cast<unsigned char>(unbase64(src[i + 0])),
```

Comment:
Question: Is there a way to make this less ugly?

Reply:
Is the compiler emitting warnings about the `static_cast`s? They shouldn’t be strictly necessary, so I think you could omit them?

(You can also omit the 4 in `... txt[4] = {` – the compiler infers the correct length itself, and it can lead to surprising results when the actual length and the length of the initializer list mismatch)

Reply:
The `txt[4]`, (i'll change `txt` for a decent name) is avoidable, but the static cast is not, just explodes.

```
../src/base64-inl.h:71:7: error: non-constant-expression cannot be narrowed from type 'int8_t' (aka 'signed char') to 'unsigned char' in initialize
r list [-Wc++11-narrowing]                                                                                                                         
      unbase64(src[i + 1]),     
```